test/testutil/provider.c

```diff
@@ -15,9 +15,13 @@ int test_get_libctx(OSSL_LIB_CTX **libctx, OSSL_PROVIDER **default_null_prov,
                     const char *config_file,
                     OSSL_PROVIDER **provider, const char *module_name)
 {
-    if ((*libctx = OSSL_LIB_CTX_new()) == NULL) {
-        opt_printf_stderr("Failed to create libctx\n");
-        goto err;
+    OSSL_LIB_CTX *new_libctx = NULL;
+
+    if (libctx != NULL) {
+        if ((new_libctx = *libctx = OSSL_LIB_CTX_new()) == NULL) {
```

Comment:
Not quite sure what you were trying to do here with this change?
Maybe if you had of done
````
new_libctx = OSSL_LIB_CTX_new();
     ....
     if (new_libctx != NULL)
         *libctx = new_libctx;
     return 1;
err:
   OSSL_LIB_CTX_free(new_libctx);
````
Otherwise I wouldnt change it.

Reply:
new_libctx only needs to be created if libctx is not NULL, otherwise the default context is used. This seemed the easiest way to allow the later calls to use a NULL libctx when loading providers into the default context, as per the comment I added in testutil.h.